### PR TITLE
Changed window reload to observable

### PR DESCRIPTION
**Pull request by KamilMatejuk**
Removed window reload when buying/upgrading building/buying cell. Instead added observable to be triggered after these actions and `city-detail.component` listens and reloads user resources.

### CODE REVIEW

File: client/src/app/city-detail/city-detail.component.ts [hunk 1]
```diff
@@ -25,10 +25,18 @@ export class CityDetailComponent implements OnInit {
...
-  constructor(private auth: AuthService) {}
+  constructor(private auth: AuthService, private city: CityService) {}
 
   ngOnInit(): void {
+    // user data
     this.userId = this.auth.GetUserData().id;
+    // user resources and production
+    this.getAllData();
+    // refreash signal
+    this.city.refreashSignal.subscribe((data) => this.getAllData());
```
Comment (jerry-sky) [inline client/src/app/city-detail/city-detail.component.ts]: Why not have a `UserService` or something that handles _all_ of user’s data? Store it in a `BehaviorSubject` that can be subscribed for changes and can hold its current value.

Then, all components that need to use any user data can just subscribe to it and update the displayed information whenever an update comes — the observable (here `BehaviorSubject`) handles the notifications about changes of the data.

File: client/src/app/services/city.service.ts [hunk 2]
```diff
@@ -1,12 +1,15 @@
...
+import { Observable, Subject } from 'rxjs';
 import { BuildingType } from '../../../../model/building-type';
 import { BackendService } from './backend.service';
 
 @Injectable({
   providedIn: 'root',
 })
 export class CityService {
+  refreashSignalSource = new Subject<string>();
+  refreashSignal = this.refreashSignalSource.asObservable();
```
Comment (jerry-sky) [inline client/src/app/services/city.service.ts]: Also, I’m pretty sure it’s spelt “refresh”.

**Review by jerry-sky — CHANGES_REQUESTED** [review]

**Comment by KamilMatejuk** [comment]
I think this is what you meant, now all data is stored in `BehaviorSubject` and handled by `UserService`.

**Review by jerry-sky — APPROVED** [review]
It still could be vastly improved, but it works, so I guess it’s alright.

**Pull request merged**